Add retry check for Sidekiq

loed-idzinga opened the pull request:
Hi there! 

In our company we are also checking the retry queue if any of the jobs go over the `DEFAULT_RETRY_CHECK` limit (for Sidekiq it defaults to 25), so we can take a look to them manually. Thought maybe I could share the code?

Our approach was to create a custom provider, so if this approach is okay I'll continue and write some specs.

lbeder commented:
This is great! Could you just make sure that every rubocop style issues are fixed (see here: https://github.com/lbeder/health-monitor-rails/actions/runs/3853885912/jobs/6578427525)?

loed-idzinga commented:
@lbeder I've updated the code 👍

lbeder commented:
Thanks, @loed-idzinga! We're almost there. There are still very minor rubocop offenses (see here: https://github.com/lbeder/health-monitor-rails/actions/runs/3910525936/jobs/6692671069) and I've left you a comment in the PR itself.

lbeder commented on spec/support/providers.rb:
```diff
@@ -80,4 +84,10 @@ def stub_sidekiq_queue_size_failure(queue_name = 'default')
+
+  def stub_sidekiq_over_retry_limit_failure
+    retry_set = instance_double(Sidekiq::RetrySet)
+    allow(retry_set).to receive(:select).and_return(["item" => { retry_count: 21 }])
```
Perhaps something like `::HealthMonitor::Providers::Sidekiq::Configuration::DEFAULT_RETRY_CHECK + 1` instead?

loed-idzinga replied:
Good catch! 👍

loed-idzinga commented:
@lbeder Thanks for the review, I've updated the code with the suggestions. 

I've moved the `let(:..)` statements around to fix `RSpec/MultipleMemoizedHelpers` Rubocop cop. Let me know if you agree with this change.
